src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java

```diff
@@ -52,7 +53,15 @@ private StudentProfileAttributes extractProfileData() {
         if ("".equals(editedProfile.nationality)) {
             editedProfile.nationality = getRequestParamValue("existingNationality");
         }
-        editedProfile.gender = getRequestParamValue(Const.ParamsNames.STUDENT_GENDER);
+
+        String value = Const.ParamsNames.STUDENT_GENDER;
+
+        if ("male".equals(value) || "female".equals(value) || "other".equals(value)) {
+            editedProfile.gender = Gender.valueOf(getRequestParamValue(Const.ParamsNames.STUDENT_GENDER));
+        } else {
+            editedProfile.gender = Gender.none;
```

Comment:
This does not really make sense and adds "none" unnecessarily. Why not just "if not male/female, it should be other"?

Reply:
Sir in case if we put an invalid parameter for gender = "/hfshh" then In this case whether it Should be other or It will give the error . As you told that if not male/female then other .

Reply:
Under normal circumstance (e.g. from the UI it would have been restricted to just male/female/other) it should not happen. It is fine to force invalid values to other.

Reply:
Okk sir